### PR TITLE
[JENKINS-46899] fixed support for core versions >= 2.64

### PR DESCRIPTION
**Pull request by daspilker**

### CODE REVIEW

File: src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiExtension.groovy [hunk 1]
```diff
@@ -169,11 +169,18 @@ class JpiExtension {
...
+                    jenkinsTest("org.jenkins-ci.main:jenkins-war:${v}:war-for-test")
+                }
+
                 jenkinsTest("org.jenkins-ci.main:jenkins-test-harness:${testHarnessVersion}")
                 jenkinsTest("org.jenkins-ci.main:ui-samples-plugin:${uiSamplesVersion}",
-                        "org.jenkins-ci.main:jenkins-war:${v}:war-for-test",
                         'junit:junit-dep:4.10')
             }
+
+            if (new VersionNumber(this.coreVersion) >= new VersionNumber('2.64')) {
```
Comment (rhencke) [inline src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiExtension.groovy]: If desired, you could consolidate this in a single check, as:
```groovy
if (new VersionNumber(this.coreVersion) < new VersionNumber('2.64')) {
    project.dependencies.add('jenkinsTest', "org.jenkins-ci.main:jenkins-war:${v}:war-for-test")
} else {
    project.configurations.jenkinsTest.extendsFrom(project.configurations.jenkinsWar)
}
```

**Comment by KrystianMarek** [comment]
Sorry, it looks like it is still broken.

`buildscript {
    repositories {
        mavenLocal()
        mavenCentral()
        maven {
            name 'jenkins'
            delegate.url('http://repo.jenkins-ci.org/releases/')
        }
    }
    dependencies {
        classpath 'org.jenkins-ci.tools:gradle-jpi-plugin:0.24.0'
    }
}

// Apply the java plugin to add support for Java
plugins {
    id 'org.jenkins-ci.jpi' version '0.24.0'
    id 'groovy'
    id 'idea'
}`

`FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':testRuntimeClasspath'.
> Could not find jenkins-war-war-for-test.jar (org.jenkins-ci.main:jenkins-war:2.87).
  Searched in the following locations:
      https://repo.jenkins-ci.org/public/org/jenkins-ci/main/jenkins-war/2.87/jenkins-war-2.87-war-for-test.jar

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 26s
8 actionable tasks: 8 executed
`

**Comment by casz** [comment]
@daspilker @KrystianMarek I have created a Jira issue that covers the same issue: https://issues.jenkins-ci.org/browse/JENKINS-47988

**Comment by Argelbargel** [comment]
@daspilker: i can confirm, that @KrystianMarek is right. I think the cause is the transitive dependency in jenkins-test-harness. I you look at the pom of [jenkins-test-harness:2.0](https://github.com/jenkinsci/jenkins-test-harness/blob/jenkins-test-harness-2.0/pom.xml) it contains the classifier war-for-test which seems to cause the problem even though the jpi-plugin does not have a direct dependency to the war-for-test anymore

**Comment by casz** [comment]
@KrystianMarek #105 should do the trick indeed 👍

**Pull request merged**